storage/libradb/src/lib.rs

```diff
@@ -687,18 +687,20 @@ impl DbReader for LibraDB {
             .with_label_values(&["get_account_state_with_proof"])
             .start_timer();
 
-        if version <= ledger_version {
-            warn!(
-                "The queried version {} should be equal to or older than ledger version {}.",
-                version, ledger_version
-            );
-        }
-        let latest_version = self.get_latest_version()?;
-        if ledger_version <= latest_version {
-            warn!(
-                "The ledger version {} is greater than the latest version currently in ledger: {}",
-                ledger_version, latest_version
-            )
+        ensure!(
+            version <= ledger_version,
+            "The queried version {} should be equal to or older than ledger version {}.",
+            version,
+            ledger_version
+        );
+        {
```

Comment:
curious why you nested / scoped this.

Reply:
`latest_version` is for this warning purpose only, I'm trying to prevent it being (mis)used in the logic below.